vnext/Microsoft.ReactNative/RedBox.cpp

```diff
@@ -151,8 +166,83 @@ struct RedBox : public std::enable_shared_from_this<RedBox> {
         m_errorMessageText.Text(Microsoft::Common::Unicode::Utf8ToUtf16(message));
         m_errorStackText.Text(Microsoft::Common::Unicode::Utf8ToUtf16(stack));
         return;
+      } else if (json.count("type") && json["type"] == "InternalError") {
+        auto message = json["message"].asString();
+        m_errorMessageText.Text(Microsoft::Common::Unicode::Utf8ToUtf16(message));
+
+        if (IsMetroBundlerError(message, json["type"].asString())) {
+          xaml::Documents::Hyperlink link;
+          link.NavigateUri(Uri(MAKE_WIDE_STR(METRO_TROUBLESHOOTING_URL)));
+          xaml::Documents::Run linkRun;
+
+          linkRun.Text(Microsoft::Common::Unicode::Utf8ToUtf16(METRO_TROUBLESHOOTING_URL));
+          link.Foreground(
+              xaml::Media::SolidColorBrush(winrt::Windows::UI::ColorHelper::FromArgb(0xff, 0xff, 0xff, 0xff)));
+          link.Inlines().Append(linkRun);
+          xaml::Documents::Run normalRun;
+          normalRun.Text(Microsoft::Common::Unicode::Utf8ToUtf16(json["type"].asString() + (" ─ See ")));
+          m_errorStackText.Inlines().Append(normalRun);
+          m_errorStackText.Inlines().Append(link);
+        } else {
+          m_errorStackText.Text(Microsoft::Common::Unicode::Utf8ToUtf16(json["type"].asString()));
+        }
+        return;
       }
     } catch (...) {
+      std::string doctype = "<!DOCTYPE HTML>";
+      if (boost::istarts_with(m_errorInfo.Message, doctype)) {
+        auto webView = xaml::Controls::WebView(xaml::Controls::WebViewExecutionMode::SameThread);
+
+        winrt::hstring content(
+            Microsoft::Common::Unicode::Utf8ToUtf16(m_errorInfo.Message.substr(doctype.length()).c_str()));
+
+        webView.HorizontalAlignment(xaml::HorizontalAlignment::Stretch);
+        webView.VerticalAlignment(xaml::VerticalAlignment::Stretch);
+        webView.MinWidth(400);
+
+        m_stackPanel.Children().Clear();
+        m_stackPanel.Children().Append(webView);
+
+        auto dispatcher = winrt::Windows::System::DispatcherQueue::GetForCurrentThread();
```

Comment:
>winrt::Windows::System::DispatcherQueue::GetForCurrentThread(); [](start = 26, length = 63)

Thanks for changing this to DispatcherQueue!  Strange... I was literally 5 seconds away from suggesting this when I saw your latest revision.